pandas/tseries/offsets.py

```diff
@@ -896,7 +896,15 @@ def apply(self, other):
 
             # adjust by business days first
             if bd != 0:
-                skip_bd = BusinessDay(n=bd)
+                if isinstance(self, _CustomMixin):  # GH 30593
+                    skip_bd = CustomBusinessDay(
+                        n=bd,
+                        weekmask=getattr(self, "weekmask"),
```

Comment:
Do you need the getattr if this is true? Couldn't we just called `self.weekmask`, `self.holidays`, etc... instead?

Reply:
Actually I was asking the exact same question when I wrote this. I saw you guys use object.__setattr__ in the __init__ method of _CustomMixin so I thought that was the paradigm here.

Anyway I have updated the code to use self.xxx directly.